### DISCUSSION TITLE
[cli] Disable checking for updates on NPM after every command

### DISCUSSION THREAD

**Discussion by rirze** [aws/aws-cdk#23857] [Q&A]
A flag (e.g. `--no-check-update`) to disable CDK from checking for updates to the CDK library on NPM. 

### Use Case

In order of least importance to most serious:

1. A mild inconvenience when waiting for commands to complete. Takes a noticeable second to finish when I most concerned about my deployment. If I've already seen the update notification already that day, the repeated checks/notifications are annoying.
2. Hangs and breaks in offline environments. For example, deploying from a machine that is closed from the outside world-wide-web causes the CDK cli to perform its operations but then hang for about 10 seconds after. At the end of this period, CDK crashes, complaining about not being able to view CDK's version online and compare it to the local installation. For offline security postures, this version check is unnecessary during deployment time.

### Proposed Solution

Add a flag that disables checking for version on NPM. I'm sure a single if statement somewhere that could implement this solution.

### Other

If there is a simple way to patch my local installation to disable version checking by NPM, please let me know so I can apply it in the meantime.

Thanks!

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by philvarner** [comment]
+1 to this -- this is particularly annoying now that 2.0 is out, and I get a warning about a (possibly-breaking) upgrade on every command I run.

**Comment by dominikschubert** [comment]
These update prompts can be skipped by setting the environment variable `CDK_DISABLE_VERSION_CHECK`.

Example:
```bash
CDK_DISABLE_VERSION_CHECK=1 cdk context
```

CDK now keeps track of these update notices and other warnings in  `~/.cdk/cache` and you can acknowledge warnings so that they won't appear again.
